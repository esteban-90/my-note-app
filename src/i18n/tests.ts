import type { Resource } from 'i18next'
import { createInstance } from 'i18next'
import { initReactI18next } from 'react-i18next'
import { ns, supportedLngs } from './app'

const resources: Resource = {}
export const i18nInstance = createInstance()

void (async () => {
  for (const n of ns) {
    for (const lng of supportedLngs) {
      resources[lng] ??= {}
      resources[lng] = {
        ...resources[lng],
        [n]: await import(`../../public/translations/${lng}/${n}.json`),
      }
    }
  }
})()

i18nInstance.use(initReactI18next).init({
  ns,
  supportedLngs,
  fallbackLng: supportedLngs[0],
  lng: supportedLngs[0],
  react: { useSuspense: false },
  resources,
})
