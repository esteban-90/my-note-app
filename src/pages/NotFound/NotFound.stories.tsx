import type { ComponentStory, ComponentMeta } from '@storybook/react'
import { NotFound } from './NotFound'

export default {
  title: 'Pages/Not Found',
  component: NotFound,
  parameters: {
    reactRouter: {
      routePath: '/test',
    },
  },
} as ComponentMeta<typeof NotFound>

const Template: ComponentStory<typeof NotFound> = () => <NotFound />

export const NotFoundExample = Template.bind({})
