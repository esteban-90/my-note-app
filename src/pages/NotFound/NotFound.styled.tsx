import styled from '@emotion/styled'
import { centered } from '@/styles'

export const Wrapper = styled.div`
  ${centered}
  flex-direction: column;
  color: var(--color-main);
  height: 85%;

  & h2 {
    font-size: 1.5rem;
  }

  & svg {
    animation-duration: 5s;
  }
`
