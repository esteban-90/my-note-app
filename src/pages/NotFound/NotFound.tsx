import type { FC } from 'react'
import { Link } from 'react-router-dom'
import { useTranslation } from 'react-i18next'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Button } from '@/components'
import { Wrapper } from '@styled/NotFound'

/**
 * Page to display if the user tries to go to a page that doesn't exist either because it corresponds
 * to a note previously removed or the id passed by parameter is invalid.
 * @returns The page.
 */

export const NotFound: FC = () => {
  const { t } = useTranslation('not-found')
  const pageHeading = t('page.heading')
  const goBackTitle = t('buttons.go-back.title')

  return (
    <Wrapper>
      <h2>{pageHeading}</h2>
      <FontAwesomeIcon icon='face-sad-tear' size='5x' fade />
      <Button icon='house' component={Link} to='/' title={goBackTitle} style={{ bottom: '1.5rem' }} />
    </Wrapper>
  )
}
