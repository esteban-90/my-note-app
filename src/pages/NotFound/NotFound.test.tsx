import { MemoryRouter, Routes, Route } from 'react-router-dom'
import { screen, render, configure } from '@/tests'
import { NotFound } from './NotFound'

describe(NotFound.name, () => {
  let asFragment: () => DocumentFragment

  const UI = (
    <MemoryRouter>
      <Routes>
        <Route path='*' element={<NotFound />} />
      </Routes>
    </MemoryRouter>
  )

  beforeAll(() => {
    configure({ throwSuggestions: true })
  })

  beforeEach(() => {
    ;({ asFragment } = render(UI))
  })

  it('should match snapshot', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render the title', () => {
    const notFoundHeading = screen.getByRole('heading', { name: /^page not found$/i })
    expect(notFoundHeading).toBeInTheDocument()
  })

  it('should render a link to home page', () => {
    const homeLink = screen.getByRole('link', { name: /^back to home$/i })
    expect(homeLink).toBeInTheDocument()
    expect(homeLink).toHaveAttribute('href', '/')
  })
})
