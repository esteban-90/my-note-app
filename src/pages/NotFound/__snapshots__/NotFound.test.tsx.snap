// Vitest Snapshot v1

exports[`NotFound > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1n2qccw"
  >
    <h2>
      Page not found
    </h2>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-face-sad-tear fa-fade fa-5x "
      data-icon="face-sad-tear"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 256C0 397.4 114.6 512 256 512s256-114.6 256-256S397.4 0 256 0S0 114.6 0 256zm240 80c0-8.8 7.2-16 16-16c45 0 85.6 20.5 115.7 53.1c6 6.5 5.6 16.6-.9 22.6s-16.6 5.6-22.6-.9c-25-27.1-57.4-42.9-92.3-42.9c-8.8 0-16-7.2-16-16zm-80 80c-26.5 0-48-21-48-47c0-20 28.6-60.4 41.6-77.7c3.2-4.4 9.6-4.4 12.8 0C179.6 308.6 208 349 208 369c0 26-21.5 47-48 47zM303.6 208c0-17.7 14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32s-32-14.3-32-32zm-128 32c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
        fill="currentColor"
      />
    </svg>
    <a
      class="css-1qofjy"
      href="/"
      style="bottom: 1.5rem;"
      title="Back to home"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-house fa-inverse fa-xl "
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
          fill="currentColor"
        />
      </svg>
    </a>
  </div>
</DocumentFragment>
`;
